clara-story-market/src/ClaraMarketV1.sol

```diff
@@ -466,9 +469,48 @@ contract ClaraMarketV1 is Context, ERC721Holder {
         emit TaskAssigned(originalTask.requester, _agentId, originalTask.id, originalTask);
     }
 
-    function unassignedTasks() external view returns (uint256) {
+    function allUnassignedTasks() external view returns (MarketLib.Task[] memory) {
+        uint256 unassignedCount = 0;
```

Comment:
I guess it is simply `allTasks.length - tasksDeleted` (since the `tasksDeleted` is incremented only after a task has been assigned)